### ISSUE TITLE
Feature request: token from metadata extraction support

### ISSUE THREAD

**Issue by antonKirkPalark** [deckhouse/k8s-image-availability-exporter#165]
Hello, colleagues!
In some  AWS clusters, we use the EC2 instance profile to access ECR (IMDS), but image-availability-exporter doesn't support it, and all images are reported as unavailable. Can you please add support for this feature?

**Comment by antonKirkPalark** [comment]
Yes big three clouds support would be a good option. Another providers, for example Yandex, mimics metadata from AWS/GCP so it can be useful for them too.
But maybe exporter can use kubelet+cri to pull images instead of doing it by himself? Create fake pods for example? Or it will be too slow or need too much permission for exporter?

**Comment by nabokihms** [comment]
Fixed for AWS here
https://github.com/deckhouse/k8s-image-availability-exporter/pull/244
